tests/scripts/thread-cert/Cert_7_1_04_BorderRouterAsRouter.py

```diff
@@ -108,6 +115,44 @@ def test(self):
             if addr[0:10] == '2001:2:0:1' or addr[0:10] == '2001:2:0:2':
                 self.assertTrue(self.nodes[LEADER].ping(addr))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        ROUTER = pv.vars['ROUTER']
+        MED = pv.vars['MED']
+        SED = pv.vars['SED']
+        _rpkts = pkts.filter_wpan_src64(ROUTER)
+
+        # Step 3: The DUT MUST send properly formatted MLE Advertisements
+        # The DUT MUST send a CoAP Server Data Notification message
+        # with the server’s information (Prefix, Border Router) to the Leader
+        _rpkts.filter_mle_cmd(MLE_CHILD_ID_REQUEST).must_next()
+        _rpkts.filter_mle_cmd(MLE_ADVERTISEMENT).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, ROUTE64_TLV} == set(p.mle.tlv.type))
+        _rpkts.filter_coap_request(SVR_DATA_URI).must_next()
+        _rpkts_med = _rpkts.copy()
+        _rpkts_sed = _rpkts.copy()
+
+        # Step 4: Automatically transmits a 2.04 Changed CoAP response to the DUT
+        # Step 5: The DUT MUST send a multicast MLE Data Response
+        _rpkts.filter_mle_cmd(MLE_DATA_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, LEADER_DATA_TLV, NETWORK_DATA_TLV, ACTIVE_TIMESTAMP_TLV} == set(p.mle.tlv.
+                                                                                                           type))
+
+        # Step 6: The DUT MUST send a Child Update Response to MED_1
+        _rpkts_med.filter_wpan_dst64(MED).filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, MODE_TLV, ADDRESS_REGISTRATION_TLV} <= set(p.mle.tlv.type))
+
+        # Step 7: The DUT MUST send an MLE Child Update Request to SED_1
+        _rpkts_sed.filter_wpan_dst64(SED).filter_mle_cmd(MLE_CHILD_UPDATE_REQUEST).must_next().must_verify(
+            lambda p: {Ipv6Addr('2001:2:0:1::')} == set(p.thread_nwd.tlv.prefix))
+
+        # Step 8: SED_1 send its configured global address to the DUT
+        # Step 9: The DUT MUST send a Child Update Response to SED_1
+        _rpkts_sed.filter_wpan_dst64(SED).filter_mle_cmd(MLE_CHILD_UPDATE_RESPONSE).must_next().must_verify(
+            lambda p: {SOURCE_ADDRESS_TLV, MODE_TLV, ADDRESS_REGISTRATION_TLV} <= set(p.mle.tlv.type))
```

Comment:
Can we verify the global addresses in these steps?

Reply:
already verify extaddrs

Reply:
Actually I meant the configured global addresses with prefix 2001:2:0:1:: or 2001:2:0:2::

Reply:
Referring to the test plan, only needs to check the 3 tlvs and the added registration addresses configured by the Child in step 8 in current commit